zio/zngio/parser.go

```diff
@@ -85,18 +85,31 @@ func zngParseContainer(builder *zcode.Builder, typ zng.Type, b []byte) ([]byte,
 // zngParseField parses the given byte array representing any value
 // in the zng format.
 func zngParseField(builder *zcode.Builder, typ zng.Type, b []byte) ([]byte, error) {
-	if b[0] == leftbracket {
-		return zngParseContainer(builder, typ, b)
-	}
+	var err error
+	var index int
 	if len(b) >= 2 && b[0] == '-' && b[1] == ';' {
-		if zng.IsContainerType(typ) {
+		if zng.IsContainerType(typ) || zng.IsUnionType(typ) {
 			builder.AppendContainer(nil)
 		} else {
 			builder.AppendPrimitive(nil)
 		}
 		return b[2:], nil
 	}
-
+	if utyp, ok := typ.(*zng.TypeUnion); ok {
+		b, typ, index, err = utyp.SplitZng(b)
+		if err != nil {
+			return nil, err
+		}
+		builder.BeginContainer()
+		defer builder.EndContainer()
+		var a [8]byte
+		n := zcode.EncodeCountedVarint(a[:], int64(index))
```

Comment:
Should this be zcode.EncodeCountedUvarint?

Reply:
+1